src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/ComAwareEventInfoTests.Windows.cs

```diff
@@ -22,7 +22,7 @@ public class DispAttributeClass
             public void Event() { }
         }
 
-        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
```

Comment:
Do all these tests really work on Nano Server?

Reply:
No idea. There is no obvious pattern for which of these tests are or aren't supported on nano server. The only limitation I am aware of on nano server is the lack of STA, which none of these should require. My intent here is to let them run and actually see what is broken. Agreed it is dual purpose. Alternative approaches are welcome. If we want to keep the limitation the name of the exclusion becomes impossibly annoying:  `IsBuiltInComEnabledAndNotWindowsNano`.

Reply:
> If we want to keep the limitation the name of the exclusion becomes impossibly annoying: IsBuiltInComEnabledAndNotWindowsNano

If we end up needing something like that, it wouldn't be the first one. The usual solution is to declare a static property on the test class itself and use `[Conditional(nameof(TheAnnoyingProperty))]` (i.e. we don't infect PlatformDetection class with such unique snowflake condition but keep it local to the test).

Reply:
I figured out why the Windows nanoserver was the issue. It seems the COM server we were expecting to exist for a lot of testing isn't available on WIndows nanoserver because it is an `IDispatch` type - blech. I created a COM server in managed and manually register it for in-proc COM using a `ModuleInitializer`. This means we can avoid the opaque dependency - yay. It does introduce a bit more code, but I think in the long run this is far better than being dependent on OS provided services that can come and go.

/cc @jkoritzinsky @elinor-fung